pkg/util/validation/exporter/exporter.go

```diff
@@ -35,7 +70,25 @@ func NewOverridesExporter(defaultLimits *Limits, tenantLimits TenantLimits) *Ove
 			[]string{"limit_name"},
 			nil,
 		),
+		logger:            log,
+		subserviceWatcher: services.NewFailureWatcher(),
+	}
+
+	if config.Ring.Enabled {
+		oeRing, err := newRing(config.Ring, log, registerer)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create ring/lifecycler")
+		}
+		exporter.ring = oeRing
+		manager, err := services.NewManager(oeRing.lifecycler, oeRing.client)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create service manager")
+		}
+		exporter.subserviceManager = manager
 	}
+
+	exporter.Service = services.NewBasicService(exporter.starting, exporter.running, exporter.stopping)
+	return exporter, nil
 }
 
 func (oe *OverridesExporter) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
Is there any existing unit test we can improve to test the ring too? Don't want to make things too complicated, but a unit test would be great.

Reply:
I gave it a try, maybe I should try to cover the ring itself more as well?